ext/mysqlnd/config9.m4

```diff
@@ -24,7 +24,9 @@ if test "$PHP_MYSQLND" != "no" || test "$PHP_MYSQLND_ENABLED" = "yes"; then
 
 
   if test "$PHP_MYSQLND_COMPRESSION_SUPPORT" != "no"; then
-    PKG_CHECK_MODULES([ZLIB], [zlib])
+    PKG_CHECK_MODULES([ZLIB], [zlib], [
+      AC_DEFINE(HAVE_ZLIB, 1, "ZLIB support")
+    ])
     PHP_EVAL_LIBLINE($ZLIB_LIBS, MYSQLND_SHARED_LIBADD)
     PHP_EVAL_INCLINE($ZLIB_CFLAGS)
     AC_DEFINE([MYSQLND_COMPRESSION_WANTED], 1, [Enable compressed protocol support])
```

Comment:
This looks unnecessarily complicated. I'd instead replace `MYSQLND_COMPRESSION_WANTED` here with `MYSQLND_COMPRESSION_ENABLED` and then drop the block at https://github.com/php/php-src/blob/d237989f33c3483dc824af695949bc37fb33c173/ext/mysqlnd/mysqlnd.h#L49-L51

There's no reason why we need both MYSQLND_COMPRESSION_WANTED and HAVE_ZLIB here. (Keep in mind that PKG_CHECK_MODULES is going to hard error if zlib is not found.)